### PR TITLE
[MPS] Fix SDPA vector kernel mask offset for partially broadcast masks

### PR DESCRIPTION
**Pull request by Isalia20**
Reproducer:
```
import torch
import torch.nn.functional as F

torch.manual_seed(0)
q = torch.randn(2, 4, 8, 64)
mask = torch.zeros(2, 1, 8, 8)
mask[0, 0, 4:, :4] = float("-inf")

y_cpu = F.scaled_dot_product_attention(q, q, q, attn_mask=mask)
y_mps = F.scaled_dot_product_attention(
    q.to("mps"), q.to("mps"), q.to("mps"), attn_mask=mask.to("mps")
)
print((y_mps.cpu() - y_cpu).abs().max().item())
# gives 0.0961458683013916
```

### CODE REVIEW

File: aten/src/ATen/native/mps/operations/Attention.mm [hunk 1]
```diff
@@ -256,12 +256,15 @@
                   scale_factor);
 
       if (has_mask) {
+        auto mask = mask_.value();
         int nd = mask_.value().dim();
```
Comment (Skylion007) [inline aten/src/ATen/native/mps/operations/Attention.mm]: This value() access is redundant now. Just use the mask reference from the line above

**Review by Skylion007 — APPROVED** [review]

File: aten/src/ATen/native/mps/operations/Attention.mm (hunk 1 above)
Comment (Isalia20) [reply aten/src/ATen/native/mps/operations/Attention.mm]: woops, missed that. Thanks, fixed
